internal/providers/vmware/vmware_amd64.go

```diff
@@ -56,3 +49,39 @@ func FetchConfig(logger *log.Logger, _ *resource.HttpClient) (types.Config, repo
 	logger.Debug("config successfully fetched")
 	return util.ParseConfig(logger, decodedData)
 }
+
+func fetchRawConfig(logger *log.Logger) (config, error) {
+	info := rpcvmx.NewConfig()
+
+	ovfEnv, err := info.String("ovfenv", "")
+	if err == nil && ovfEnv != "" {
```

Comment:
It looks like this is conflating the error-case and the missing-case. Should this perhaps be split into two err-check+return first and then empty-check+logic blocks?